kaldi import/export -- adding basic tests for kaldi import/export

jtrmal opened the pull request:
I'm fairly new to pytest -- please let me know if it's merge-worthy or should we re-done in some other way.
I will continue on adding more tests/checks

jtrmal commented:
The fails are just flac missing. Not important for now
y.

pzelasko commented:
Looks good! Do you think we can get away with a smaller number of files, like 2-3 FLACs? It would make the clone/download size much smaller.

csukuangfj commented:
How about generating the sound files dynamically with sox?

jtrmal commented:
I removed extra files that I do not have tests for yet (feature-files related mostly) and implemented the sox suggestion (which is a great thinking!). @pzelasko  ready for review/merge if you like

jtrmal commented:
ugh, not ready yet, sorry!

jtrmal commented:
I think this is pretty much all the test cases I was able to figure out. let me know.

csukuangfj commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+import pytest
+import os.path
```
People usually use `import os`, I think.
After that, you can use `os.path` in the code.

Also, people usually sort the imports within a block alphabetically, i.e.,
```python3
import contextlib
import os
import pytest
```

csukuangfj commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+import lhotse.utils
+from lhotse.utils import is_module_available
+
+FIXTURE_PATH = Path(os.path.realpath(__file__)).parent / "fixtures"
```
```suggestion
FIXTURE_PATH = Path(__file__).resolve().parent / "fixtures"
```

and you can use
```python
Path('xxx').is_file()  # or Path('xxx').exists()
```
in the following and remove `import os.path`.

csukuangfj commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+    return recording, supervision
+
+
+@pytest.mark.xfail(reason="multi fail recordings not supported yet")
```
```suggestion
@pytest.mark.xfail(reason="multi file recordings not supported yet")
```

pzelasko commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+def test_multi_file_recording(tmp_path, multi_file_recording):
+    with working_directory(tmp_path):
+        lhotse.kaldi.export_to_kaldi(
+            multi_file_recording[0], multi_file_recording[1], ".", None, False
```
can you use keyword args here (and other places)? I'm not sure what ".", None, and False correspond to when reading the test

pzelasko commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+
+
+@pytest.mark.xfail(reason="Import ordering is somehow not working for now")
+def test_get_duration_failurex(monkeypatch):
```
will fix imports (for future self: `from augmentation import *` is broken in top-level `__init__.py`)

pzelasko commented on test/test_kaldi_dirs.py:
```diff
@@ -0,0 +1,305 @@
+
+
+@contextlib.contextmanager
+def working_directory(path):
```
nice readability trick

pzelasko reviewed (commented):
Thanks a lot, looks good to me! Can you address the kwarg comment?

jtrmal commented:
thanks for the comments, will resolve it tomorrow.
y.

On Wed, Apr 27, 2022 at 7:07 PM Fangjun Kuang ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860306212>:
>
> > @@ -0,0 +1,305 @@
>
> +import pytest
>
> +import os.path
>
>
> People usually use import os, I think.
> After that, you can use os.path in the code.
>
> Also, people usually sort the imports within a block alphabetically, i.e.,
>
> import contextlib
> import os
> import pytest
>
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860307584>:
>
> > @@ -0,0 +1,305 @@
>
> +import pytest
>
> +import os.path
>
> +import contextlib
>
> +from pathlib import Path
>
> +
>
> +import lhotse
>
> +import lhotse.utils
>
> +from lhotse.utils import is_module_available
>
> +
>
> +FIXTURE_PATH = Path(os.path.realpath(__file__)).parent / "fixtures"
>
>
> ⬇️ Suggested change
>
> -FIXTURE_PATH = Path(os.path.realpath(__file__)).parent / "fixtures"
>
> +FIXTURE_PATH = Path(__file__).resolve().parent / "fixtures"
>
>
> and you can use
>
> Path('xxx').is_file()  # or Path('xxx').exists()
>
> in the following and remove import os.path.
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860307892>:
>
> > +            lhotse.SupervisionSegment(
>
> +                id="lbi-1272-135031-0000-B",
>
> +                recording_id="lbi-1272-135031-0000",
>
> +                start=0.0,
>
> +                duration=10.885,
>
> +                channel=1,
>
> +                text="SOMETHING ELSE",
>
> +                speaker="lbi-1272-135031-B",
>
> +                gender="f",
>
> +            ),
>
> +        ]
>
> +    )
>
> +    return recording, supervision
>
> +
>
> +
>
> ***@***.***(reason="multi fail recordings not supported yet")
>
>
> ⬇️ Suggested change
>
> ***@***.***(reason="multi fail recordings not supported yet")
>
> ***@***.***(reason="multi file recordings not supported yet")
>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lhotse-speech/lhotse/pull/687#pullrequestreview-955620712>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUKYX3BM7JGWNQDWWIDJGTVHHCDTANCNFSM5UPLE2LQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jtrmal commented:
yeah, will do
y.

On Wed, Apr 27, 2022 at 9:20 PM Piotr Żelasko ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Thanks a lot, looks good to me! Can you address the kwarg comment?
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860367189>:
>
> > +                duration=10.885,
> +                channel=1,
> +                text="SOMETHING ELSE",
> +                speaker="lbi-1272-135031-B",
> +                gender="f",
> +            ),
> +        ]
> +    )
> +    return recording, supervision
> +
> +
> ***@***.***(reason="multi fail recordings not supported yet")
> +def test_multi_file_recording(tmp_path, multi_file_recording):
> +    with working_directory(tmp_path):
> +        lhotse.kaldi.export_to_kaldi(
> +            multi_file_recording[0], multi_file_recording[1], ".", None, False
>
> can you use keyword args here (and other places)? I'm not sure what ".",
> None, and False correspond to when reading the test
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860368124>:
>
> > +        out = lhotse.kaldi.load_kaldi_data_dir(fixture_path, 16000, 0.01, replace, 1)
> +
> +    lhotse_dir = "lhotse"
> +    if replace:
> +        lhotse_dir += "-" + replace
> +
> +    with working_directory(fixture_path / lhotse_dir):
> +        recording_set = lhotse.RecordingSet.from_jsonl("recordings.jsonl.gz")
> +        supervision_set = lhotse.SupervisionSet.from_jsonl("supervisions.jsonl.gz")
> +
> +    assert out[0] == recording_set
> +    assert out[1] == supervision_set
> +
> +
> ***@***.***(reason="Import ordering is somehow not working for now")
> +def test_get_duration_failurex(monkeypatch):
>
> will fix imports (for future self: from augmentation import * is broken
> in top-level __init__.py)
> ------------------------------
>
> In test/test_kaldi_dirs.py
> <https://github.com/lhotse-speech/lhotse/pull/687#discussion_r860368772>:
>
> > +def test_multi_file_recording(tmp_path, multi_file_recording):
> +    with working_directory(tmp_path):
> +        lhotse.kaldi.export_to_kaldi(
> +            multi_file_recording[0], multi_file_recording[1], ".", None, False
> +        )
> +
> +
> +def test_multi_channel_recording(tmp_path, multi_channel_recording):
> +    with working_directory(tmp_path):
> +        lhotse.kaldi.export_to_kaldi(
> +            multi_channel_recording[0], multi_channel_recording[1], ".", None, False
> +        )
> +
> +
> ***@***.***
> +def working_directory(path):
>
> nice readability trick
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lhotse-speech/lhotse/pull/687#pullrequestreview-955703389>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACUKYX762QGOYOYOO3BUSKDVHHRUHANCNFSM5UPLE2LQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

jtrmal commented:
@pzelasko @csukuangfj  I addressed your feedback
